arrow-core-data/src/main/kotlin/arrow/core/Either.kt

```diff
@@ -926,6 +926,60 @@ sealed class Either<out A, out B> : EitherOf<A, B> {
       } catch (t: Throwable) {
         fe(t.nonFatalOrThrow()).left()
       }
+
+    /**
+     * The resolve function can resolve any suspended function that yields an Either into one type of value.
+     *
+     * @param f the function that needs to be resolved.
+     * @param ifSuccess the function to apply if [f] yields a success of type [A].
+     * @param ifDomainError the function to apply if [f] yields a domain error of type [E].
+     * @param ifSystemFailure the function to apply if [f] throws a [Throwable].
+     * @param ifHandlingCaseThrows the function to apply if the [ifSuccess], [ifDomainError] or [ifSystemFailure] function throw a [Throwable].
+     * Throwing any [Throwable] in the [ifHandlingCaseThrows] function will render the [resolve] function nondeterministic.
+     * @param ifUnrecoverableState the function to apply if [resolve] is in an unrecoverable state.
+     * @return the result of applying the [resolve] function.
+     */
+    suspend fun <E, A, B> resolve(
+      f: suspend () -> Either<E, A>,
+      ifSuccess: suspend (a: A) -> Either<Throwable, B>,
```

Comment:
```suggestion
      success: suspend (a: A) -> Either<Throwable, B>,
```

Reply:
Changed the naming suggestions using the IDE instead of committing the suggestion from GitHub. That way I could be sure that all references are updated as well.